h3/src/client.rs

```diff
@@ -65,14 +66,20 @@ where
     }
 }
 
-impl<S, C> RequestStream<FrameStream<S>, Bytes, Connection<C>>
+pub struct RequestStream<S, C>
+where
+    C: quic::Connection<Bytes>,
```

Comment:
I believe this generic is to allow internal `connection::RequestStream` to know if it can `send_request` or `accept`. If that's the only thing, it'd be more convenient I think for users if we could just remove this generic, and not worry about enforcing that internally, only via the separate public `RequestStream` types.

Reply:
Yeah, this wasn't obvious how to remove this at the time. Turns out it is useless :sweat_smile: